service/interpreter/workflowImpl.go

```diff
@@ -763,18 +763,31 @@ func invokeStateExecute(
 		// this is not a problem because the signalWithStart will be very fast and highly available
 		unifiedClient := env.GetUnifiedClient()
 
-		var workflowId string
+		var parentWfId, currentWorkflowId, newWorkflowId string
+
+		response, err := unifiedClient.DescribeWorkflowExecution(context.Background(), executionContext.WorkflowId, "", nil)
```

Comment:
we should get the firstRunId from workflowContext.
Basically, temporal workflow provides the runtime context via context. (similarly iWF workflowState has Context). 
The runtime context has access to this runId directly without making an API call.

Reply:
I'm not sure if this is what you meant by workflowContext. I added FirstRunID to WorkflowInfo interface and use it this way:
```
provider.GetWorkflowInfo(ctx).FirstRunID
```

I think that's what you meant